flexmeasures/ui/views/analytics.py

```diff
@@ -535,6 +535,16 @@ def make_power_figure(
     else:
         title = "Electricity production from %s" % resource_display_name
 
+    resolution_str = "?"
+    if data.index.freq is not None:
+        resolution_str = time_utils.freq_label_to_human_readable_label(
+            data.index.freqstr
+        )
+    elif "resolution" in session:
```

Comment:
Since our integration of timely-beliefs `BeliefsDataFrames`, I wouldn't consider this a decent fallback anymore. The `event_resolution` is where we store the resolution of the data we serve. If you want to keep this fallback (in case the `event_resolution` is None), I would log a warning, too. Because the fallback shouldn't be needed. Or is this to support the case of empty data? I think that even if the data is empty, the `event_resolution` should actually still have been set to the requested resolution already.